### PR TITLE
Upgrade to SDK v3.4

### PR DESCRIPTION
**Pull request by mody**
Also works with the old SDK

BFW-5632

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,14 +3,10 @@ FROM ubuntu:22.04
 ENV TZ=Etc/UTC
 RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
 
-RUN apt-get update
-RUN apt-get -y install bash git cmake python3 python3-pip python3-virtualenv python3-cryptography python3-future python3-click python3-serial python3-wheel python3-pyparsing python3-pyelftools
+RUN apt-get update && apt-get -y install bash git cmake python3 python3-pip python3-virtualenv python3-cryptography python3-future python3-click python3-serial python3-wheel python3-pyparsing python3-pyelftools python-is-python3 libncurses-dev flex bison gperf
```
Comment (Tomcus) [inline Dockerfile]: Just a nitpick, but I think this long script is bit unreadable. Can you please do the `&& \<newline>` to format it better.

File: CMakeLists.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@
 # CMakeLists in this exact order for cmake to work correctly
 cmake_minimum_required(VERSION 3.5)
 
+set(LANGUAGE, C)
```
Comment (Tomcus) [inline CMakeLists.txt]: Do we need specify the language here? I would like to have some copied files from the main printer repo to make sure they are 1:1 and I would like to write them in c++.

**Review by Tomcus — APPROVED** [review]
LGTM

File: CMakeLists.txt (hunk 2 above)
Comment (mody) [reply CMakeLists.txt]: Fixed in 6945f3a06adbb5a5fbb37f87395c202c4c687915

File: Dockerfile (hunk 1 above)
Comment (mody) [reply Dockerfile]: Fixed in c2d03d9b8b299ef0c71d9928c87b7066ebb20e23

**Pull request merged**